### PR TITLE
improve coverage and gas snapshot

### PR DESCRIPTION
**Pull request by 3esmit**
## Description

WIP to complete coverage and make gas snapshots meaningful

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [ ] Added natspec comments?
- [x] Ran `forge snapshot`?
- [x] Ran `pnpm lint`?
- [x] Ran `pnpm gas-report`?

### CODE REVIEW

**Comment by 3esmit** [comment]
Missing coverage:

//transfer tests
testInvalidDestinationTransfer
testTransferNoBalance
testTransferDisabled
testTransferFromDisabled
testTransferOverflow
testTransferControllerRejected
testTransferControllerNotContract
//allowance test
testNoAllowance
testApproveTransferDisabled
testApproveControllerRejected
testAllowanceAlreadySet
testResetAllowance
testApproveAndCall
//Clone tests
testFutureParentSnapshot
testBalanceOfAtClone
testTotalSupplyAtClone
testGenerateTokensSupplyOverflow
testGenerateTokensBalanceOverflow
testDestroyTokensNotEnoughSupply
testDestroyTokensNotEnoughBalance
testGetValueNull
testGetValueTooEarlyFromBlock
testControllerZero

**Comment by 0x-r4bbit** [comment]
I will give this another review once the other cases are done, but I just wanted to say: great job man!!

So good to see all these tests. Well done!

**Comment by 3esmit** [comment]
Some tests remaining in the checkmarks seem not possible to do, or to reach, for now. 

We should merge this as it is now, as it covers at least 88.42% of the contract. 
I noticed that coverage says some parts are not fully reached, when I am certain they are really reached, but need to confirm later.

File: test/MiniMeToken.t.sol [hunk 1]
```diff
@@ -1,12 +1,27 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.19;
 
-import { Test } from "forge-std/Test.sol";
+import { Test, console } from "forge-std/Test.sol";
```
Comment (0x-r4bbit) [inline test/MiniMeToken.t.sol:4]: This import can be removed

File: test/MiniMeToken.t.sol [hunk 2]
```diff
@@ -46,21 +64,110 @@ contract MiniMeTokenTest is Test {
     }
 }
 
+contract AcceptingController is TokenController {
+    receive() external payable { }
```
Comment (0x-r4bbit) [inline test/MiniMeToken.t.sol:68]: Might be worth looking into so the linter doesn't complain about empty blocks in `receive()` (or `fallback()`)

File: test/MiniMeToken.t.sol [hunk 3]
```diff
@@ -115,16 +427,253 @@ contract AllowanceTest is MiniMeTokenTest {
...
+            }
+            //if we want a zero-length slice let's just return a zero-length array
+            default {
+                tempBytes := mload(0x40)
+
+                mstore(0x40, add(tempBytes, 0x20))
+            }
+        }
+
+        return tempBytes;
```
Comment (0x-r4bbit) [inline test/MiniMeToken.t.sol:676]: We migtht want to consider moving these contracts used for tests out of the test file as they bloat the test file and they contain quite a bit of logic in them.

**Review by 0x-r4bbit — APPROVED** [review]
Massive work @3esmit 
There are some comments we should address, but will create separate issues for that.

**Pull request merged**